lib/cookie.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+class Cookie
+  attr_reader :setcookie_string
+  def initialize(str)
+    @setcookie_string = str
+  end
+
+  def to_s
+    "ORA_WX_SESSION='#{ora_wx_session_id}';portal=#{portal_id}"
+  end
+
+  def ora_wx_session_id
+    setcookie_string.split('SESSION="')[1].split('";').first
```

Comment:
This seems like quite a brittle approach to extract the information you want, and it also heavily disguises what you're actually trying to access. Is there not a better way to parse this?